### PR TITLE
Fix pagination when passing a query to jobs.list

### PR DESCRIPTION
**Pull request by aripollak**
* Also fix bug in VCR sensitive data filtering by removing `<>` from censored string so we don't mistake it for an XML tag. This was causing parts of tags to be completely removed from the cassette. The broken files had not been checked in, so tests on Travis CI were still passing.

### CODE REVIEW

File: lib/tableau_api/connection.rb [hunk 1]
```diff
@@ -3,28 +3,28 @@ class Connection
...
-    def api_get_collection(path, collection, *args)
+    def api_get_collection(path, collection, page_number: 1, page_size: 100, **kwargs)
       Enumerator.new do |enum|
-        args[0] = {} unless args[0]
-        page_size = (args[0].delete(:page_size) { 100 }).to_i
-        page_number = (args[0].delete(:page_number) { 1 }).to_i
-
         loop do
-          uri = URI::HTTP.build(path: "/#{path}", query: URI.encode_www_form(**args[0], pageSize: page_size, pageNumber: page_number)).request_uri
+          query = kwargs.fetch(:query, '')
```
Comment (mattyb) [inline lib/tableau_api/connection.rb:22]: So does this only support passing query as a string? Does httparty generally support a hash? But then it does tableau-incompatible URL encoding?

File: spec/fixtures/vcr_cassettes/users.yml [hunk 2]
```diff
@@ -107,19 +107,19 @@ http_interactions:
...
 - request:
     method: get
-    uri: http://TABLEAU_HOST/api/3.1/sites/f31923d2-dea4-4470-843a-9fc999d240aa/users?pageNumber=1&pageSize=100
+    uri: http://TABLEAU_HOST/api/3.1/sites/fb50c166-f809-44e0-995e-2cf56ceffbc0/users?pageNumber=1&pageSize=100
     body:
       encoding: US-ASCII
       string: ''
     headers:
       X-Tableau-Auth:
-      - uRTWEYOxQRi2XsUiy58Wiw|xz3JBXTLpZ8d83WgENB1xTPJ42mua5Hp
```
Comment (mattyb) [inline spec/fixtures/vcr_cassettes/users.yml:122]: should user agent have changed a few lines down?

File: spec/fixtures/vcr_cassettes/projects.yml [hunk 3]
```diff
@@ -44,28 +44,28 @@ http_interactions:
...
-    uri: http://TABLEAU_HOST/api/3.1/sites/f31923d2-dea4-4470-843a-9fc999d240aa/projects
+    uri: http://TABLEAU_HOST/api/3.1/sites/fb50c166-f809-44e0-995e-2cf56ceffbc0/projects
     body:
       encoding: UTF-8
       string: <tsRequest><project name="test_project" description=""/></tsRequest>
     headers:
       Content-Type:
       - application/xml
       X-Tableau-Auth:
-      - OXRCK7BwRs2Aef3k4qwD3Q|P1F5qfqYEuHaijNTIsejpLf7vcMCJvhF
```
Comment (mattyb) [inline spec/fixtures/vcr_cassettes/projects.yml:68]: checking another example...no user-agent change

File: spec/resources/jobs_spec.rb [hunk 4]
```diff
@@ -28,5 +28,12 @@ def find_or_publish_workbook(name)
       jobs = client.jobs.list
       expect(jobs.to_a.last.keys).to include('createdAt', 'id', 'jobType', 'priority', 'status')
     end
+
+    it 'can filter jobs in a site' do
+      workbook = find_or_publish_workbook('test')
+      client.workbooks.refresh(workbook_id: workbook['id'])
+      jobs = client.jobs.list(query: 'filter=jobType:eq:refresh_extracts,status:eq:Success')
```
Comment (mattyb) [inline spec/resources/jobs_spec.rb:35]: would it be a cleaner interface if we could do
```
client.jobs.list(query: { filter: 'jobType:eq:refresh_extracts,status:eq:Success' })
```
?

**Review by mattyb — CHANGES_REQUESTED** [review]
Couple thoughts

File: lib/tableau_api/connection.rb (hunk 1 above)
Comment (aripollak) [reply lib/tableau_api/connection.rb:22]: That's correct on all counts.

File: spec/fixtures/vcr_cassettes/users.yml (hunk 2 above)
Comment (aripollak) [reply spec/fixtures/vcr_cassettes/users.yml:122]: I didn't regenerate all the cassettes after the user-agent change, just jobs.yml. Should I update everything?

File: spec/resources/jobs_spec.rb (hunk 4 above)
Comment (aripollak) [reply spec/resources/jobs_spec.rb:35]: I would like to accept a hash instead, but I feel like the workarounds for the colon-encoding issue would make this not so clean. We could still support hashes later, but for now this only supports strings.

**Review by mattyb — APPROVED** [review]
LGTM

**Pull request merged**